### PR TITLE
Added gprc REST gateway unit tests

### PR DESCRIPTION
**Pull request by Ekliptor**
This PR has a couple of changes related to the REST gateway. I split them into different commits so they can be reviewed easier.

The main change:
- added gprc REST gateway unit tests
to test this just run: `go test gw_test.go` as mentioned in the README

other changes:
- updated grpc-gateway to v1.16.0
- updated swagger generation proto command to latest openapi v2 command
- added `--go_opt=paths=source_relative` proto generate command so that files are generated in correct relative path. This should have been added when `option go_package="github.com/gcash/bchd/bchrpc/pb"` was added to the `proto` file. Otherwise the proto compiler will create a subdirectory with that path `github.com/gcash/...` instead of just putting the generated code into `pb`
- removed duplicate `bchrpc.pb.go` file and deleted the `gen` folder. All auto-generated proto code is now in the BCHD `pb` folder. The files ending with `pb.go` and `gw.pb.go` must be in the same folder (and package). But having the same proto file twice issues warnings on import (as soon as we import `pb/bchrpc.pb.go` inside of `gw.go`): https://developers.google.com/protocol-buffers/docs/reference/go/faq#namespace-conflict

### CODE REVIEW

**Comment by Ekliptor** [comment]
I added JSON schema for property validation.
Issues:
- mint baton TXID sometimes null. Is this ok? `token_metadata.89.nft1_group.mint_baton_txid: Invalid type. Expected: string, given: null`
- SLP address empty in `GetTransaction()` and `GetAddressUnspentOutputs()`. Also I set the min length to 55 characters? `outputs.3.slp_token.address: String length must be greater than or equal to 55`
- BCH address empty in `GetTransaction()` and `GetAddressUnspentOutputs()`. Also I set the min length to 54 characters? `transaction.outputs.3.address: String length must be greater than or equal to 54`
- in `GetBip44HdAddress()` the pub_key response is always 44 characters. I set min length to 40?
- `document_hash` and `document_url` are the only props of tokens that can be null? I set them to `"type": ["string", "null"],`

Also if this REST API becomes an important part of BCHD:
For the next release I recommend writing most of those JSON Schema definitions directly in a `.proto` file. For example add a 2nd `bchrpc-rest.proto` file with `options` describing JSON schema and import it into the main `bchrpc.proto`: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto#L18

File: bchrpc/pb/bchrpc.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.25.0
-// 	protoc        v3.12.4
+// 	protoc        v3.3.0
```
Comment (jcramer) [inline bchrpc/pb/bchrpc.pb.go:4]: Need to use the same version as bchd master 3.12.4

**Review by jcramer — CHANGES_REQUESTED** [review]
For now, this is the only issue I see that may impede the upstream merge effort.

File: bchrpc/pb/bchrpc.pb.go (hunk 1 above)
Comment (Ekliptor) [reply bchrpc/pb/bchrpc.pb.go]: I just updated it to `v3.14.0` which is newer than in BCHD master branch. If they want to keep the exact version on merge they can just run `regen.sh`

**Comment by jcramer** [comment]
I ported these over as a squashed commit, and updated as needed.

**Pull request closed without merging**